src/OCRNetEngine.cpp

```diff
@@ -23,11 +24,27 @@ OCRNetEngine::OCRNetEngine(const std::string& engine_path, const std::string& di
     {
         mDict.emplace_back("CTCBlank");
     }
-    else
+    else if (mDecodeMode == Attention)
     {
         mDict.emplace_back("[GO]");
         mDict.emplace_back("[s]");
-    };
+    }
+    else if (mDecodeMode == CLIP)
+    {
+        mDict.emplace_back("[E]");
+        mClipCharTrainDict.emplace_back("[E]");
+        std::string clip_charset_train = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ!\"#$%&'()*+,-./:;<=>?@[\\]^_`{|}~";
```

Comment:
Why the charset is hard coded ?

Reply:
CLIP4STR uses different charset for training and test, this hard code is the training charset ,and test charset will be read from "character_list" file as previous method.

Reply:
So why cannot train and test use the same charset and be configurable from `character_list`?

Reply:
it's doable to train and test with same charset, but I haven't launch such a experiment.  In original CLIP4STR, it is designed to train and test with different charset

Reply:
I'm not familiar with CLIP4STR, but from the decoding logic, I don't see there is any blocker to use the same train charset in test. If a filter is needed, one can implement it based on the output of the nvOCDR.